code/modules/vending/custom.dm

```diff
@@ -0,0 +1,235 @@
+	if (new_slogan)
+		slogan_list += new_slogan
+		last_slogan = world.time + rand(0, slogan_delay)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/vending/custom/on_deconstruction(disassembled)
+	unbuckle_all_mobs(TRUE)
+	var/turf/current_turf = get_turf(src)
+	if(current_turf)
+		explosion(src, devastation_range = -1, light_impact_range = 3)
+
+/obj/machinery/vending/custom/vend(list/params, mob/living/user, list/greyscale_colors)
+	. = FALSE
+	if(!isliving(user))
+		return
+	var/obj/item/dispensed_item = locate(params["ref"])
+	if(!dispensed_item)
+		return
+
+	var/obj/item/card/id/id_card = user.get_idcard(TRUE)
+	if(!id_card || !id_card.registered_account || !id_card.registered_account.account_job)
+		balloon_alert(user, "no card found!")
+		flick(icon_deny, src)
+		return
+
+	/// Charges the user if its not the owner
+	var/datum/bank_account/payee = id_card.registered_account
+	if(!compartmentLoadAccessCheck(user))
+		if(!payee.has_money(dispensed_item.custom_price))
+			balloon_alert(user, "insufficient funds!")
+			return
+		/// Make the transaction
+		payee.adjust_money(-dispensed_item.custom_price, , "Vending: [dispensed_item]")
+		linked_account.adjust_money(dispensed_item.custom_price, "Vending: [dispensed_item] Bought")
+		linked_account.bank_card_talk("[payee.account_holder] made a [dispensed_item.custom_price] \
+		cr purchase at your custom vendor.")
+		/// Log the transaction
+		SSblackbox.record_feedback("amount", "vending_spent", dispensed_item.custom_price)
+		log_econ("[dispensed_item.custom_price] credits were spent on [src] buying a \
+		[dispensed_item] by [payee.account_holder], owned by [linked_account.account_holder].")
+		/// Make an alert
+		var/ref = REF(user)
+		if(last_shopper != ref || purchase_message_cooldown < world.time)
+			speak("Thank you for your patronage [user]!")
+			purchase_message_cooldown = world.time + 5 SECONDS
+			last_shopper = ref
+
+	/// Remove the item
+	use_energy(active_power_usage)
+	if(user.CanReach(src) && user.put_in_hands(dispensed_item))
+		to_chat(user, span_notice("You take [dispensed_item.name] out of the slot."))
+	else
+		to_chat(user, span_warning("[capitalize(format_text(dispensed_item.name))] falls onto the floor!"))
+	return TRUE
+
+/obj/item/vending_refill/custom
+	machine_name = "Custom Vendor"
+	icon_state = "refill_custom"
+	custom_premium_price = PAYCHECK_CREW
+
+/obj/machinery/vending/custom/unbreakable
+	name = "Indestructible Vendor"
+	resistance_flags = INDESTRUCTIBLE
+
+/obj/machinery/vending/custom/greed //name and like decided by the spawn
+	icon_state = "greed"
+	icon_deny = "greed-deny"
+	panel_type = "panel4"
+	max_integrity = 700
+	max_loaded_items = 40
+	light_mask = "greed-light-mask"
+	custom_materials = list(/datum/material/gold = SHEET_MATERIAL_AMOUNT * 5)
+
+/obj/machinery/vending/custom/greed/Initialize(mapload)
+	. = ..()
+	//starts in a state where you can move it
+	set_panel_open(TRUE)
+	set_anchored(FALSE)
+	add_overlay(panel_type)
```

Comment:
Lol why is this manually adding the panel overlay, does `set_panel_open` not handle updating appearance?